test/system_SUITE.erl

```diff
@@ -0,0 +1,461 @@
+-module(system_SUITE).
+-compile([export_all]).
+
+-include_lib("common_test/include/ct.hrl").
+-include_lib("amqp_client/include/amqp_client.hrl").
+
+-define(SERVER_REJECT_CLIENT, {tls_alert,"unknown ca"}).
+
+all() ->
+    [
+     library,
+     invasive_SSL_option_change,
+     validation_success_for_AMQP_client,
+     validation_failure_for_AMQP_client,
+     whitelisted_certificate_accepted_from_AMQP_client_regardless_of_validation_to_root,
+     removed_certificate_denied_from_AMQP_client,
+     installed_certificate_accepted_from_AMQP_client,
+     whitelist_directory_DELTA,
+     replaced_whitelisted_certificate_should_be_accepted,
+     ensure_configuration_using_binary_strings_is_handled
+    ].
+
+suite() ->
+    [{timetrap, {seconds, 25}}].
+
+%% -------------------------------------------------------------------
+%% Testsuite setup/teardown.
+%% -------------------------------------------------------------------
+
+init_per_suite(Config) ->
+    rabbit_ct_helpers:log_environment(),
+    Config1 = rabbit_ct_helpers:set_config(Config, [
+        {rmq_nodename_suffix, ?MODULE}
+      ]),
+    Config2 = rabbit_ct_helpers:set_config(Config1, [{rmq_extra_tcp_ports, [tcp_port_amqp_tls_extra]}]),
+    rabbit_ct_helpers:run_setup_steps(Config2,
+      rabbit_ct_broker_helpers:setup_steps() ++
+      rabbit_ct_client_helpers:setup_steps()).
+
+end_per_suite(Config) ->
+    rabbit_ct_helpers:run_teardown_steps(Config,
+      rabbit_ct_client_helpers:teardown_steps() ++
+      rabbit_ct_broker_helpers:teardown_steps()).
+
+init_per_group(_, Config) ->
+    Config.
+
+end_per_group(_, Config) ->
+    Config.
+
+init_per_testcase(Testcase, Config) ->
+    ok = build_directory_tree(friendlies(Config)),
```

Comment:
If you generate certificate for each testcase, you should name the directory or the certificates after the testcase name, available in `Testcase`. And during teardown, do not delete the generated data: it allows for later inspection during debugging.

To be even safer in the case of testcases in groups, you can use the following helper to generate a directory name for each testcase:
```erlang
rabbit_ct_helpers:config_to_testcase_name(Config, Testcase)
```

The result is of the form `group/subgroup/testcase`.